daemon: Use rpc for "version" command

ndorf opened the pull request:
Fixes #5994. To be useful, #6017 and #6015 are also needed.

moneromooo-monero commented on src/daemon/rpc_command_executor.cpp:
```diff
@@ -2364,4 +2364,39 @@ bool t_rpc_command_executor::set_bootstrap_daemon(
+    }
+    else
+    {
+        tools::success_msg_writer() << res.version;
```
This lets a daemon controlled by a third party put something like this in the version string:
WARNING: your wallet is old and vulnerable to <insert bullshit here>. Please upgrade ASAP to https://pwn-me-now.com/fake-monero.tar.gz
Maybe some sanitization should be done on the version string first.

ndorf replied:
How about now?

ndorf commented:
Updated this to only print the version string if it matches a rather strict regex, and added a unit test to ensure that the current `MONERO_VERSION` passes, among other things.

I considered a more lenient check, but any remotely useful one would still need the unit test, at which point we might as well check very strictly.

ndorf commented:
Minor update; no reason to create an `std::string` for `fail_message`

ndorf commented:
Another minor update; just making a temporary `regex` object is probably better than having a static one.
